flutter-mentorship-t4/mentorship_project
# feature (Cart) :sparkles: Finishing Cart Logic

# Pull Request: Cart Logic Implementation 

## Jira Ticket: [DIVA-17](https://nz-dev.atlassian.net/jira/software/projects/DIVA/boards/1?selectedIssue=DIVA-17)

## Description

I have completed the implementation of the cart logic in our Flutter app. This PR includes the state management for adding and removing items from the cart, handling quantity updates, and calculating the total price.

## How to Test

1. Open the Flutter app.
2. Navigate to a product details page.
3. Add the product to the cart and verify that:
   * The item appears in the cart with the correct initial quantity.
   * The total price is calculated correctly.
4. Increment and decrement the quantity of an item in the cart:
   * Verify that the quantity updates correctly.
   * Check that the total price adjusts accordingly.
5. Remove an item from the cart:
   * Confirm that the item is removed from the cart list.
   * Ensure the total price is recalculated correctly.
6. Add multiple items to the cart and perform the above operations to test edge cases.
7. Close and reopen the app to verify that the cart state persists.


## Screenshots

<img src="https://github.com/user-attachments/assets/bc9a1b1b-cb6e-40f2-810b-ba2e3f5b2394" width="200"/>
<img src="https://github.com/user-attachments/assets/f157b525-d1e0-4216-9271-e062b9aeb5b2" width="200"/>
<img src="https://github.com/user-attachments/assets/f2aed503-3b5d-484f-a718-c615264ae554" width="200"/>
<img src="https://github.com/user-attachments/assets/807ea211-7cc0-451c-8cc3-506383229166" width="200"/>
<img src="https://github.com/user-attachments/assets/15edacc0-6495-434f-99e3-9f5ca2a5aff8" width="200"/>
<img src="https://github.com/user-attachments/assets/89b61ee0-9d30-463a-9dd3-c4c1b7673435" width="200"/>
<img src="https://github.com/user-attachments/assets/5badf8de-8afe-4e88-aae8-f2fd2cac1566" width="200"/>


## Additional Notes

- I used shared prefrences to store the cart data locally and didnt use the API 


[DIVA-17]: https://nz-dev.atlassian.net/browse/DIVA-17?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
